test/molecule-role/molecule/vms/tests/test_receiver_topology.py

```diff
@@ -191,3 +191,42 @@ def wait_for_components():
         ) is not None
 
     util.wait_until(wait_for_components, 120, 3)
+
+
+def test_host_topology(host):
+    agent_hosts = AnsibleRunner(os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('agent_linux_vm')
+
+    for hostname in agent_hosts:
+        def wait_for_components():
+            url = "http://localhost:7070/api/topic/sts_topo_process_agents?offset=0&limit=1000"
+            data = host.check_output("curl \"%s\"" % url)
+            json_data = json.loads(data)
+            with open("./topic-topo-process-agents-topology-{}.json".format(hostname), 'w') as f:
```

Comment:
@zandrewitte just a question why do we write the file when it is not required anywhere except for debugging i think.